tests/Teams/ActionBasedMessagingExtension-FetchTask/Bots/ActionBasedMessagingExtensionFetchTaskBot.cs

```diff
@@ -17,5 +20,110 @@ protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivi
         {
             await turnContext.SendActivityAsync(MessageFactory.Text($"echo: {turnContext.Activity.Text}"), cancellationToken);
         }
+
+        protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionFetchTaskAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionQuery query, CancellationToken cancellationToken)
+        {
+            var reply = MessageFactory.Text("OnTeamsMessagingExtensionFetchTaskAsync MessagingExtensionQuery: " + JsonConvert.SerializeObject(query));
+            await turnContext.SendActivityAsync(reply, cancellationToken);
+
+            return new MessagingExtensionActionResponse
+            {
+                Task = new TaskModuleContinueResponse
+                {
+                    Value = new TaskModuleTaskInfo()
+                    {
+                        Card = new Attachment
+                        {
+                            Content = GetTaskModuleAdaptiveCard(),
+                            ContentType = AdaptiveCard.ContentType,
+                        },
+                        Height = 450,
+                        Width = 450,
+                        Title = "Task Module Example",
+                    },
+                },
+            };
+        }
+
+        protected override async Task<MessagingExtensionActionResponse> OnTeamsMessagingExtensionSubmitActionAsync(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action, CancellationToken cancellationToken)
+        {
+            var reply = MessageFactory.Text("OnTeamsMessagingExtensionSubmitActionAsync MessagingExtensionAction: " + JsonConvert.SerializeObject(action));
+            await turnContext.SendActivityAsync(reply, cancellationToken);
```

Comment:
Does this work if I use the action extension in a channel that the bot does not have access to?

Part of me thinks this scenario is fine, and another part of me thinks we should perhaps break out the `OnTeamsMessagingExtensionSubmitActionAsync` into a separate Scenario.

Reply:
No, sending to a conversation in which the bot is not installed throws an Unauthorized.  

Do you think we should either 1) remove these lines that send the message, or 2) add a try/catch and ignore the exception ?

I think having OnTeamsMessagingExtensionSubmitActionAsync in this Scenario helps differentiate it from the non-task module Messaging Extension Scenario  But, I don't feel strongly about this.  We should discuss which Scenarios we want to cover which functionality (there is a multitude of different flows possible, and the interactions vary depending on a variety of factors).